File: Cliente_cpp/src/modelos/Celular.cpp
#include "modelos/Celular.h"
#include <iostream>

Celular::Celular(int id,
                 const std::string& nome,
                 const std::string& descricao,
                 double preco,
                 int quantidade_estoque,
                 const std::string& marca,
                 const std::string& modelo)
    : Produto(id, nome, descricao, preco, quantidade_estoque),
      marca(marca),
      modelo(modelo) {}

TipoProduto Celular::getTipo() const {
    return TipoProduto::CELULAR;
}

void Celular::mostrar() const {
    Produto::mostrar();
    std::cout << "Marca: " << marca
              << " | Modelo: " << modelo << std::endl;
}

std::string Celular::getMarca() const { return marca; }
std::string Celular::getModelo() const { return modelo; }

void Celular::setMarca(const std::string& marca) {
    this->marca = marca;
}

void Celular::setModelo(const std::string& modelo) {
    this->modelo = modelo;
}

File: Servidor_java/Modelos/Celular.java
package Servidor_java.Modelos;
public class Celular extends Produto{

    private String marca;
    private String modelo;
    

    public Celular(int id,String nome,String descricao,double preco,int estoque,String marca,String modelo){

        super(id,nome,descricao,preco,estoque);

        this.marca = marca;
        this.modelo = modelo;
    }

    public String getMarca(){return marca;}
    public String getModelo(){return modelo;}

    public void setMarca(String marca){this.marca = marca;}
    public void setModelo(String modelo){this.modelo = modelo;}

    public String toString() {
        return "Celular [id=" + getId() + ", nome=" + getNome() + ", preco=" + getPreco() + ", estoque=" + getEstoque() + "Marca="+ marca + "modelo="+ modelo +"]";
    }

}